freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -529,16 +583,24 @@ private void checkIfClientStillUsable(FreeIpaClientException e) throws FreeIpaCl
 
     @VisibleForTesting
     void validateParameters(String accountId, String actorCrn, Set<String> environmentCrnFilter,
-            Set<String> userCrnFilter, Set<String> machineUserCrnFilter) {
+            Set<String> userCrnFilter, Set<String> machineUserCrnFilter, Optional<String> deletedWorkoadUser) {
```

Comment:
I would prefer if you pass `UserSyncRequestFilter` instead of adding a new parameter.
also moving the whole validation to a separate class would be nice (`@VisibleForTesting` annotation already a sign this method should be in another class)